pkg/debug/Dockerfile

```diff
@@ -63,13 +63,54 @@ ADD https://github.com/pixel/hexedit/archive/refs/tags/1.5.tar.gz ../1.5.tar.gz
 RUN tar -C .. -xzvf ../1.5.tar.gz
 RUN ./autogen.sh && ./configure && make DESTDIR=/out install
 
+WORKDIR /usr/src
+ADD https://cdn.openbsd.org/pub/OpenBSD/OpenSSH/portable/openssh-9.8p1.tar.gz /usr/src
+RUN tar xvf openssh-9.8p1.tar.gz
+WORKDIR /usr/src/openssh-9.8p1
+COPY ssh/fix-utmpx-ipv6.patch .
+RUN git apply fix-utmpx-ipv6.patch
+RUN ./configure \
+    --prefix=/usr \
+    --sysconfdir=/etc/ssh \
+    --libexecdir=/usr/lib/ssh \
+    --mandir=/usr/share/man \
+    --with-pid-dir=/run \
+    --with-mantype=doc \
+    --disable-utmp \
+    --disable-wtmp \
+    --disable-lastlog \
+    --disable-strip \
+    --with-privsep-path=/var/empty \
+    --with-xauth=/usr/bin/xauth \
+    --with-default-path='/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin' \
+    --with-privsep-user=sshd \
+    --with-ssl-engine
+
+RUN make -j "$(getconf _NPROCESSORS_ONLN)"
+RUN make install
+RUN HARNESS_JOBS="$(getconf _NPROCESSORS_ONLN)" make tests
+
+RUN /usr/bin/install -c -m 0755 -s ssh /out/usr/bin/ssh
+RUN /usr/bin/install -c -m 0755 -s scp /out/usr/bin/scp
+RUN /usr/bin/install -c -m 0755 -s ssh-keygen /out/usr/bin/ssh-keygen
+RUN /usr/bin/install -c -m 0755 -s sshd /out/usr/sbin/sshd
+RUN /usr/bin/install -c -m 0755 -s sshd-session /out/usr/libexec/sshd-session
+RUN /usr/bin/install -c -m 4711 -s ssh-keysign /out/usr/libexec/ssh-keysign
+RUN /usr/bin/install -c -m 0755 -s ssh-pkcs11-helper /out/usr/libexec/ssh-pkcs11-helper
+RUN /usr/bin/install -c -m 0755 -s ssh-sk-helper /out/usr/libexec/ssh-sk-helper
+RUN /usr/bin/install -c -m 0755 -s sftp /out/usr/bin/sftp
+RUN /usr/bin/install -c -m 0755 -s sftp-server /out/usr/libexec/sftp-server
+RUN mkdir -p /out/etc/ssh
+COPY ssh/* /out/etc/ssh/
+RUN mkdir -p /out/var/empty
+
 # tweaking various bit
 WORKDIR /out
 COPY ssh.sh spec.sh scripts/ ./usr/bin/
-RUN mkdir -p ./etc/ssh ./root/.ssh && chmod 0700 ./root/.ssh
+RUN mkdir -p ./root/.ssh
```

Comment:
```suggestion
RUN mkdir -m 0700 -p ./root/.ssh
```

Reply:
`hadolint: SC2174 warning: When used with -p, -m only applies to the deepest directory.`

Reply:
that's exactly what you want, but it's ok if you don't want to add the ignore to this check, but at least group both `mkdir ... && chmod` on the same RUN would avoid a separated layer only because of the chmod command...

Reply:
Docker does not create separate layers anymore, so therefore I prefer the readability.

Reply:
ok, I see you have added another directory to the mkdir, so it makes more sense separate the chmod... but let's remember it's recommended by docker as good practice to keep commands packed into a single RUN whenever it's possible... regarding your question:

You don't see the layers because the Dockerfile it's using a multi-stage build approach (which is good), so layers of intermediate images are kept only in cache and put in the final container only what is really needed, but it doesn't mean intermediate layers are not created, for instance, you can build your container with docker and export the full cache like this:

```sh
docker buildx build --cache-to type=local,dest=$PWD/lcache,mode=max --output type=docker,name=lfedge/debug:test pkg/debug
```

You should see a lot of layers, those corresponding to each stage of the Dockerfile. However, if you don't use the `mode=max`, it will export only the layers of the final container, only those corresponding to commands below the "FROM scratch" in the Dockerfile, those that we see in the final container...

Reply:
I get: